examples/single_particle/demo_curvature_B.jl

```diff
@@ -0,0 +1,59 @@
+# magnetic field curvature drift
+# More theoretical details can be found in F.F.Chen's Introduction to Plasma Physics and Controlled Fusion.
+
+using TestParticle
+using TestParticle: get_gc
+using TestParticleMakie
+using OrdinaryDiffEq
+using StaticArrays
+using LinearAlgebra
+using GLMakie
+using ForwardDiff: gradient
+
+function curved_B(x)
+    # satisify ∇⋅B=0
+    # B_θ = 1/r => ∂B_θ/∂θ = 0
+    θ = atan(x[3]/(x[1]+3))
+    r = hypot(x[1]+3, x[3])
+    return SA[-1e-7*sin(θ)/r, 0, 1e-7*cos(θ)/r]
+end
+
+function zero_E(x)
+    return SA[0, 0, 0]
+end
+
+abs_B(x) = norm(curved_B(x))  # |B|
+
+# trace the orbit of the guiding center
+function trace_gc!(dx, x, p, t)
+    q, m, E, B, sol = p
+    xu = sol(t)
+    xp = @view xu[1:3]
+    gradient_B = gradient(abs_B, x)  # ∇|B|
+    Bv = B(x)
+    b = normalize(Bv)
+    v_para = (xu[4:6]⋅b).*b  # (v⋅b)b
+    v_perp = xu[4:6] - v_para
+    dx[1:3] = m*(0.5*norm(v_perp)^2+0*norm(v_para)^2)*(Bv×gradient_B)/(q*norm(Bv)^3) + (E(x)×Bv)/norm(Bv)^2 + 
+                v_para  # (m/q)*(v∥^2+0.5*v⟂^2)*(B×∇B)/B^3 + (E×B)/B^2 + v∥

```

Comment:
Maybe "para" has the same 4 chars as "perp" that looks symmetric, but usually the shorthand notation is "par". It's kind of sad that "⟂" (\perp) is not allowed to use for variables in Julia right now. "\parallel" is supported.